File: Tutor Demos/Week 5/Java/Strategy.java
public class Strategy {
    private String focus;
    public Strategy() {
        this.focus = "Attack";
    }

    public void changeStrategy() {
        if (focus.equals("Attack")) {
            focus = "Defence";
        }
        else {
            focus = "Attack";
        }
        System.out.println("Strategy changed to " + focus);
    }

    @Override
    public String toString() {
        return this.focus;
    }
}

File: Tutor Demos/Week 5/Python/strategy.py
class Strategy:
    def __init__(self):
        self.__focus = "Attack"
    
    def change_stratgy(self):
        if self.__focus == "Attack":
            self.__focus = "Defence"
        else:
            self.__focus = "Attack"
        print(f"Strategy changed to {self.__focus}")
    
    def __str__(self):
        return self.__focus